src/linux/component.rs

```diff
@@ -178,6 +179,41 @@ impl From<u8> for TermalSensorType {
     }
 }
 
+/// Check given `item` dispatch to read the right `file` with the right parsing and store data in
+/// given `component`. `id` is provided for `label` creation.
+fn fill_component(component: &mut Component, id: u32, item: &str, folder: &Path, file: &str) {
+    let hwmon_file = folder.join(file);
+    match item {
+        "type" => {
+            component.sensor_type =
+                read_number_from_file::<u8>(&hwmon_file).map(|n| TermalSensorType::from(n))
+        }
+        "input" => {
+            let temperature = get_temperature_from_file(&hwmon_file);
+            component.input_file = hwmon_file;
+            component.temperature = temperature;
+            // Max computed value taken form current temperatue if not possible to
+            // get it from `highest`.
+            if let None = component.max {
+                component.max = temperature;
+            }
+        }
+        "label" => component.label = get_file_line(&hwmon_file, 10).unwrap_or(String::new()),
+        "highest" => {
+            component.max = get_temperature_from_file(&hwmon_file).or(component.temperature);
+            component.highest_file = Some(hwmon_file);
+        }
+        "max" => component.threshold_max = get_temperature_from_file(&hwmon_file),
+        "min" => component.threshold_min = get_temperature_from_file(&hwmon_file),
+        "crit" => component.threshold_critical = get_temperature_from_file(&hwmon_file),
+        // Raspberry_pi weird comportement.
+        // TODO: Test it on a raspberry pi.
+        "" if component.label.is_empty() => component.label = format!("Component name {}", id),
+        // TODO maybe it's good to trace for being future aware of a change in hwmon temp?
```

Comment:
You can do so using the `sysinfo_debug!` macro. ;)

Reply:
Will do it it should'nt be neccessary but I did'nt have a pi to test it out so...